### PR TITLE
Updates README, jdk+node version and setup steps for android and iOS

### LINKED ISSUE #459: Readme update with latest version requirements and setup steps
### Issue Description

This issue involves updating the Java and Node.js versions in readme. Additionally, steps to set up the React Native mobile app will be documented for future reference.

## Current Behavior

Outdated documentation.

## Expected Behaviour

- Update the readme and proper steps to setup react native in rds mobile app.

## Screenshots
n/a

## Environment

- **Operating System**: [e.g., Windows, macOS, Linux]  
- **Browser (if applicable)**: [e.g., Chrome, Firefox, Safari]  
- **Version of Software/Project**: [e.g., v1.0.0]  

## Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

Thank you for contributing to our project! We appreciate your help in improving it.

📚 See [contributing instructions](https://github.com/samyakshah3008/mobile-app/blob/develop/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by Ayushsanjdev**
Date: 09-Oct-2024

Developer Name:  Ayush Sanj (Ayushsanjdev)

---

## Issue Ticket Number

Closes #459  

## Description

- This updates Volta to latest node version.
- This updates jdk version.
- steps that gets Contributors run the app on local as well as on simulators/physical devices.
- added some icons to look better on readme for titles

### Documentation Updated?

- [x] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>
image
</details>

<details>
<summary>Screenshot 2</summary>
image
</details>

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>
N/A

</details>

## Additional Notes

N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "RdsApp",
+  "name": "realdevsquad-mobile",
```
Comment (ashifkhn) [inline package.json]: revert this change please

**Review by ashifkhn — CHANGES_REQUESTED** [review]

**Review by ashifkhn — CHANGES_REQUESTED** [review]
Left one comment

File: package.json (hunk 1 above)
Comment (Ayushsanjdev) [reply package.json]: done

File: README.md [hunk 2]
```diff
@@ -1,107 +1,168 @@
...
-- Profile Screen in which we can change our profile through our phone’s gallery ✅
-- API integration in Profile Screen ✅
+## 📖 About the Project
 
-## Second Phase
-- Three Screens - Home , Goals and Profile ✅
-- Home screen Features ✅
-- Goals site Tab
-- Goals site feature will have the TODO app where you can assign the task to the member of the RDS.
+This project serves as the Real Dev Squad cross-platform mobile app, built using React Native. It aims to provide a seamless experience for our users by replicating the features of our web application.
```
Comment (shreya-mishra) [inline README.md:13]: not just replicating the web features, it will have other features as well.

File: README.md [hunk 3]
```diff
@@ -1,107 +1,168 @@
...
+- **Availability Toggle:** Switch between “Out of Office” and “Back Again” ✅
+- **Profile Screen:** Modify profile details using your phone’s gallery ✅
+- **API Integration in Profile Screen** ✅
 
-## Prerequisites
+### Second Phase
 
+- **Three Screens:** Home, Goals, and Profile ✅
+- **Home Screen Features** ✅
+- **Goals Tab:** Displays the TODO app where you can assign tasks to the members of the Real Dev Squad.
```
Comment (shreya-mishra) [inline README.md:33]: We can remove this from the readme file. Could you take other's opinions as well?

Comment (Ayushsanjdev) [reply README.md:33]: Makes sense, sure

File: README.md (hunk 2 above)
Comment (Ayushsanjdev) [reply README.md:13]: What should I add there?

File: README.md (hunk 3 above)
Comment (Ayushsanjdev) [reply README.md:33]: @ashifkhn what are your thoughts on this?

File: README.md (hunk 2 above)
Comment (Ayushsanjdev) [reply README.md:13]: Any suggestions @shreya-mishra @ashifkhn

**Comment by Ayushsanjdev** [comment]
@ashifkhn please review this once, meanwhile I update the IOS docs, if you see above shreya left some comments and I needed your opinion too.
